flickit-assessment-core/src/main/java/org/flickit/assessment/core/adapter/out/persistence/user/UserPersistenceJpaAdapter.java

```diff
@@ -19,4 +19,9 @@ public class UserPersistenceJpaAdapter implements LoadUserPort {
     public Optional<User> loadById(UUID userId) {
         return repository.findById(userId).map(UserMapper::mapToDomainModel);
     }
+
+    @Override
+    public User loadByEmail(String email) {
```

Comment:
Please return an Optional instead of returning null and performing the null check in the service.